var/spack/repos/builtin/packages/py-hail/package.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyHail(MakefilePackage):
+    """Cloud-native genomic dataframes and batch computing (Python API)"""
+
+    homepage = "https://hail.is"
+    git = "https://github.com/hail-is/hail.git"
+    # We can't use tarballs because HAIL needs to look up git commit metadata
+    # to determine its version. We could patch this, but that is not yet
+    # implemented.
+    # url = "https://github.com/hail-is/hail/archive/refs/tags/0.2.130.tar.gz"
+
+    maintainers("teaguesterling")
+    license("MIT", checked_by="teaguesterling")
+
+    version("0.2.132", commit="678e1f52b9999cb05ebf03fd360e5c4506bd6dad")
+    version("0.2.131", commit="11d9b2ff89da9ef6a4f576be89f1f06959580ea4")
+    version("0.2.130", commit="bea04d9c79b5ca739364e8c121132845475f617a")
+    version("0.2.129", commit="41126be2df04e4ef823cefea40fba4cadbe5db8a")
+
+    resource(
+        name="catch",
+        url="https://github.com/catchorg/Catch2/releases/download/v2.6.0/catch.hpp",
+        sha256="a86133b34d4721b6e1cf7171981ea469789f83f2475907b4033012577e4975fe",
+        destination="hail/src/main/resources/include/catch.hpp",
+        expand=False,
+    )
```

Comment:
Is there any reasonable way to depend on `catch2`?

Reply:
> Is there any reasonable way to depend on `catch2`?

They version it into the source tree during the installation, which seemed difficult to address with spack. The build process is rather quirky, to say the least.